chartfx-samples/src/main/java/de/gsi/chart/samples/financial/service/SCIDByNio.java

```diff
@@ -0,0 +1,298 @@
+package de.gsi.chart.samples.financial.service;
+
+import de.gsi.chart.samples.financial.dos.Interval;
+import de.gsi.chart.samples.financial.dos.OHLCVItem;
+import javafx.beans.property.DoubleProperty;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Create OHLCV from Sierra Chart SCID files
+ */
+public class SCIDByNio {
+    private FileChannel fileChannel;
+    private ByteBuffer bufferHeader;
+    private ByteBuffer bufferRecordDouble;
+    private ByteBuffer bufferRecordFloat;
+    private ByteBuffer bufferRecordULong;
+    private Calendar cal = Calendar.getInstance();
+    private int timeZone;
+
+    private String title;
+    private String symbol;
+
+    @SuppressWarnings("resource")
+    public void openNewChannel(String resource) throws IOException {
+        title = new File(resource).getName();
+        symbol = title.replaceFirst("[.][^.]+$", "");
+
+        timeZone = cal.get(Calendar.ZONE_OFFSET);
+
+        File f = new File(resource);
+        FileInputStream fis = new FileInputStream(f);
+
+        //----------------------------------
+        fileChannel = fis.getChannel();

```

Comment:
Yes, it cannot be closed in the same method because it is a opened stream for ohlc realtime processing. There is another method for closing stream in the finished/exit parts of sample demo. LGTM cannot recognize this handling by some simple checking of try-resource. The LGTM suppress tags are already added.